update gradlew to be able to build on recent java versions

With the current version of the wrapper '**gradle-4.5.1**', gradle does not support Java 11+. Therefore, it fails when trying to build making this project unusable for any Java version above 9. (Current LTS version is 11, soon 17) Correct me if I am wrong. 

Gradlew exception:
```
pi@raspberrypi:/tmp/rpi_ws281x_build $ ./gradlew assemble

FAILURE: Build failed with an exception.

* What went wrong:
Could not determine java version from '11.0.9.1'.
```

For more details: 
https://stackoverflow.com/questions/54358107/gradle-could-not-determine-java-version-from-11-0-2
and
https://www.oracle.com/java/technologies/java-se-support-roadmap.html

--- a/README.md
+++ b/README.md
@@ -3,15 +3,64 @@ rpi281x wrapper for Java using SWIG
 
 ### To build on a raspberry pi
 
-Run `src/scripts/createNativeLib.sh` to generate the SWIG java code and generate the libws2811.so native library.
+Run `src/scripts/createNativeLib.sh` to generate the SWIG java code and generate the libws2811.so native library (For a tutorial on how to install SWIG, see "Install SWIG on RaspberryPi" below).
 
 Run `./gradlew assemble` to compile the java code and create a jar containing the compile class files and the native .so file.
 
 ### To build from another machine
 
-The machine must be a linux machine, or have a bash-compatible shell like babun, cygwin, or git-bash on windows.
+Set the appropriate username, host and password in `gradle.properties`.
 
-Run `./build-native-on-remote-pi.sh`.  This will copy the project to the pi, and run the script in the previous section, and copy the .so library back to the dev machine. After that, run `.\gradlew assemble` to compile the java code and create a jar containing the compile class files and the native .so file.  The easiest way to use the jar would be to publish it to a maven repository, or your local .m2 repository using `./gradlew publishToMavenLocal` and use maven coordinates in your maven or gradle project. 
+Then, run `./gradlew buildNativeOnPi`.  Note that your RaspberryPi needs to have ssh installed as gradle will use ITS ssh client.
+
+Alternatively, you can also use the script `build-native-on-remote-pi.sh` with bash or a bash-compatible shell (like babun, cygwin, or git-bash on windows.).
+
+
+This will copy the project to the pi, and run the script in the previous section, and copy the .so library back to the dev machine. After that, run `.\gradlew assemble` to compile the java code and create a jar containing the compile class files and the native .so file.  The easiest way to use the jar would be to publish it to a maven repository, or your local .m2 repository using `./gradlew publishToMavenLocal` and use maven coordinates in your maven or gradle project. 
+
+#### Install SWIG on RaspberryPi
+
+These commands worked for me:
+1. Install [prereqs](https://github.com/swig/swig/wiki/Getting-Started#linux---ubuntu)
+```shell
+sudo apt-get install build-essential libpcre2-dev libpcre3-dev
+```
+2. Install default jdk
+```shell
+sudo apt install default-jdk-headless
+```
+
+3. Download swig tarball and unzip
+```shell
+wget http://prdownloads.sourceforge.net/swig/swig-4.1.1.tar.gz
+tar -xzvf swig-4.1.1.tar.gz
+```
+4. enter directory and start building
+```shell
+cd swig-4.1.1
+./configure && make
+```
+5. Execute an example as a check
+```shell
+cd Examples/java/class && make check
+```
+6. Install SWIG
+```shell
+cd ../../../
+sudo make install
+swig -version
+```
+```
+SWIG Version 4.1.1
+
+Compiled with g++ [x86_64-pc-linux-gnu]
+
+Configured options: +pcre
+
+Please see https://www.swig.org for reporting bugs and further information
+```
+
+For reference, the [SWIG DOCS](https://www.swig.org/Doc4.1/SWIGDocumentation.html#Preface_unix_installation).
 
 Attribution
 -----------

--- a/README.md
+++ b/README.md
@@ -1,6 +1,12 @@
 # rpi-ws281x-java
 rpi281x wrapper for Java using SWIG
 
+## TL;DR
+1. Set RaspberryPi ssh username and password in gradle.properties
+2. run `./gradlew installSwigOnPi`
+3. run `./gradlew buildNativeOnPi`
+4. run `./gradlew publishToMavenLocal`
+
 ### To build on a raspberry pi
 
 Run `src/scripts/createNativeLib.sh` to generate the SWIG java code and generate the libws2811.so native library (For a tutorial on how to install SWIG, see "Install SWIG on RaspberryPi" below).
@@ -22,9 +28,11 @@ This will copy the project to the pi, and run the script in the previous section
 
 You can try `./gradlew installSwigOnPi` to automatically execute all the steps detailed below on your pi.
 
-**NOTE:** 
+**NOTE:**
 The default-jdk-headless which this task will try to install works only on pis with ARMv7+ architecture (check with the command `cat /proc/cpuinfo`).
 The script will try to detect this and install the zulu jdk as an alternative, but this has only been tested on a raspberrypi Zero so far.
+On my RasPi Zero this task took 30min the majority of which was spent building SWIG.
+On the RasPi 3 B it took 8 mins.
 
 **So proceed with caution!**
 

--- a/README.md
+++ b/README.md
@@ -9,7 +9,7 @@ Run `./gradlew assemble` to compile the java code and create a jar containing th
 
 ### To build from another machine
 
-Set the appropriate username, host and password in `gradle.properties`.
+Set the appropriate username, host and password (or path to private key) in `gradle.properties`.
 
 Then, run `./gradlew buildNativeOnPi`.  Note that your RaspberryPi needs to have ssh installed as gradle will use ITS ssh client.
 
@@ -19,6 +19,8 @@ Alternatively, you can also use the script `build-native-on-remote-pi.sh` with b
 This will copy the project to the pi, and run the script in the previous section, and copy the .so library back to the dev machine. After that, run `.\gradlew assemble` to compile the java code and create a jar containing the compile class files and the native .so file.  The easiest way to use the jar would be to publish it to a maven repository, or your local .m2 repository using `./gradlew publishToMavenLocal` and use maven coordinates in your maven or gradle project. 
 
 #### Install SWIG on RaspberryPi
+
+You can try `./gradlew installSwigOnPi` to automatically execute all the steps detailed below on your pi.
 
 These commands worked for me:
 1. Install [prereqs](https://github.com/swig/swig/wiki/Getting-Started#linux---ubuntu)

--- a/README.md
+++ b/README.md
@@ -11,7 +11,7 @@ Run `./gradlew assemble` to compile the java code and create a jar containing th
 
 Set the appropriate username, host and password (or path to private key) in `gradle.properties`.
 
-Then, run `./gradlew buildNativeOnPi`.  Note that your RaspberryPi needs to have ssh installed as gradle will use ITS ssh client.
+Then, run `./gradlew buildNativeOnPi`.  Note that your RaspberryPi needs to have ssh installed as gradle will use *its* ssh client.
 
 Alternatively, you can also use the script `build-native-on-remote-pi.sh` with bash or a bash-compatible shell (like babun, cygwin, or git-bash on windows.).
 
@@ -21,6 +21,14 @@ This will copy the project to the pi, and run the script in the previous section
 #### Install SWIG on RaspberryPi
 
 You can try `./gradlew installSwigOnPi` to automatically execute all the steps detailed below on your pi.
+
+**NOTE:** 
+The default-jdk-headless which this task will try to install works only on pis with ARMv7+ architecture (check with the command `cat /proc/cpuinfo`).
+The script will try to detect this and install the zulu jdk as an alternative, but this has only been tested on a raspberrypi Zero so far.
+
+**So proceed with caution!**
+
+You can read more in the [pi4j docs](https://pi4j.com/documentation/java-installation/).
 
 These commands worked for me:
 1. Install [prereqs](https://github.com/swig/swig/wiki/Getting-Started#linux---ubuntu)

--- a/build.gradle
+++ b/build.gradle
@@ -1,27 +1,38 @@
 plugins {
     id 'net.researchgate.release' version '2.6.0'
-    id 'org.hidetake.ssh' version '2.11.2'
     id 'java'
     id 'maven-publish'
     id 'signing'
-}
-java {
-    sourceCompatibility = JavaVersion.VERSION_17
-}
-
-remotes {
-    rasPi {
-        host = project.findProperty("RasPiHost")
-        user = project.findProperty("RasPiUser")
-        password = project.findProperty("RasPiPassword")
-        fileTransfer = "scp"
-        jschLog = true
-    }
 }
 
 apply plugin: 'java'
 apply plugin: 'maven-publish'
 apply plugin: 'signing'
+
+java {
+    sourceCompatibility = JavaVersion.VERSION_17
+}
+repositories {
+    mavenCentral()
+}
+
+configurations { sshAntTask }
+dependencies {
+    compileOnly 'org.apache.logging.log4j:log4j-api:2.10.0'
+    runtimeOnly 'org.apache.logging.log4j:log4j-core:2.12.4'
+    sshAntTask 'org.apache.ant:ant-jsch:1.10.14'
+    testImplementation 'junit:junit:4.13.1'
+}
+
+ant.taskdef(
+        name: 'scp',
+        classname: 'org.apache.tools.ant.taskdefs.optional.ssh.Scp',
+        classpath: configurations.sshAntTask.asPath)
+
+ant.taskdef(
+        name: 'ssh',
+        classname: 'org.apache.tools.ant.taskdefs.optional.ssh.SSHExec',
+        classpath: configurations.sshAntTask.asPath)
 
 group = 'com.github.mbelling'
 
@@ -37,66 +48,71 @@ sourceSets {
         }
     }
 }
+def remBuildDir = project.findProperty("RasPiRemoteBuildDir")
 
-repositories {
-    mavenCentral()
-}
-
-dependencies {
-    compileOnly 'org.apache.logging.log4j:log4j-api:2.10.0'
-    runtimeOnly 'org.apache.logging.log4j:log4j-core:2.12.4'
-
-    testImplementation 'junit:junit:4.13.1'
-}
-
-def remoteBuildDir = project.findProperty("RasPiRemoteBuildDir")
 tasks.register('buildNativeOnPi') {
     description = 'build native lib on remote pi'
+    def host = project.findProperty("RasPiHost")
+    def un = project.findProperty("RasPiUser")
+    def pw = project.findProperty("RasPiPassword")
     doLast {
-        ssh.run {
-            session(remotes.rasPi) {
-                println("[buildNativeOnPi] - Copy all files...")
-                put from: './src/', into: remoteBuildDir
-
-                //remove windows line breaks
-                execute "cd $remoteBuildDir; tr -d '\\r' <src/scripts/createNativeLib.sh >src/scripts/createNativeLib.sh"
-                execute "cd $remoteBuildDir; chmod +x src/scripts/createNativeLib.sh"
-                //now build
-                println("[buildNativeOnPi] - Execute build script...")
-                execute "cd $remoteBuildDir; touch logs.txt; bash src/scripts/createNativeLib.sh > logs.txt 2>&1"
-
-                println("[buildNativeOnPi] - Copy generated artifacts back")
-                get from: "$remoteBuildDir/build", into: '.'
-
-                println("[buildNativeOnPi] - All Done! Try ./gradlew assemble")
-            }
-        }
-
-    }
-}
-tasks.register('finalizeBuildNative') {
-    doLast {
-        ssh.run {
-            session(remotes.rasPi) {
-                get from: "$remoteBuildDir/logs.txt", into: 'build/'
-            }
-            println("The captured log output from 'createNativeLib.sh'...")
-            def logFile = project.file("build/logs.txt")
-
-            if (logFile.exists()) {
-                // Read and print the log file contents
-                logFile.eachLine { line ->
-                    println line
-                }
-            } else {
-                println "Log file not found."
-            }
-        }
-
+        println("trying to log in to $un@$host with pw $pw")
+        ant.scp(
+                file: "src/ $un@$host:$remBuildDir",
+                password: $pw
+                host: host,
+                username: un,
+                password: pw,
+                verbose: true,
+                failonerror: true
+        )
+        ant.ssh(
+                host: host,
+                username: un,
+                password: pw,
+                command: "sudo bash $remBuildDir/src/scripts/createNativeLib.sh",
+                trust: 'yes',
+                verbose: true,
+                outputproperty: "outProp",
+                resultproperty: "resProp",
+                failonerror: true
+        )
     }
 }
 
-tasks.buildNativeOnPi.finalizedBy('finalizeBuildNative')
+tasks.register("finalizeBuild") {
+    doLast {
+
+        println("**********************************************************************")
+        println("                         LOG from remote build:")
+        println("**********************************************************************")
+        println(ant.getProperty("outProp"))
+    }
+}
+buildNativeOnPi.finalizedBy finalizeBuild
+
+
+/*
+        println("[buildNativeOnPi] - Copy all files...")
+        commandLine 'scp'
+        put from: './src/', into: remoteBuildDir
+
+        //remove windows line breaks
+        execute "cd $remoteBuildDir; tr -d '\\r' <src/scripts/createNativeLib.sh >src/scripts/createNativeLib.sh"
+        execute "cd $remoteBuildDir; chmod +x src/scripts/createNativeLib.sh"
+        //now build
+        println("[buildNativeOnPi] - Execute build script...")
+        execute "cd $remoteBuildDir; touch logs.txt; bash src/scripts/createNativeLib.sh > logs.txt 2>&1"
+
+        println("[buildNativeOnPi] - Copy generated artifacts back")
+        get from: "$remoteBuildDir/build", into: '.'
+
+        println("[buildNativeOnPi] - All Done! Try ./gradlew assemble")
+
+    }
+}*/
+
+tasks.register('copyFilesBackAfterRemoteBuild')
 
 
 tasks.register('checkSharedLibrary') {

--- a/build.gradle
+++ b/build.gradle
@@ -20,10 +20,6 @@ remotes {
     }
 }
 
-apply plugin: 'java'
-apply plugin: 'maven-publish'
-apply plugin: 'signing'
-
 group = 'com.github.mbelling'
 //flag to disable signing if keyInfo ain't present
 boolean doSigning = project.hasProperty('signing.keyId')
@@ -34,13 +30,17 @@ sourceSets {
     main {
         java {
             // Get the generated source from SWIG
-            srcDirs file('build/generatedSource/java')
+            srcDirs('build/generatedSource/java')
         }
         resources {
             // Get the native .so file
-            srcDirs file('build/nativeLib/')
+            srcDirs('build/nativeLib/')
         }
     }
+}
+
+jar {
+    from sourceSets.main.output
 }
 
 repositories {
@@ -48,8 +48,7 @@ repositories {
 }
 
 dependencies {
-    compileOnly 'org.apache.logging.log4j:log4j-api:2.10.0'
-    runtimeOnly 'org.apache.logging.log4j:log4j-core:2.12.4'
+    implementation 'org.slf4j:slf4j-api:1.7.32'
 
     testImplementation 'junit:junit:4.13.1'
 }
@@ -109,21 +108,11 @@ tasks.register('checkSharedLibrary') {
     }
 }
 
-tasks.register('sourceJar', Jar) {
-    from sourceSets.main.allJava
-}
+compileJava.dependsOn checkSharedLibrary
 
-tasks.register('javadocJar', Jar) {
-    dependsOn javadoc
-    from javadoc.destinationDir
-}
-
-build.dependsOn checkSharedLibrary
-
-artifacts {
-    archives jar
-    archives sourceJar
-    archives javadocJar
+java {
+    withJavadocJar()
+    withSourcesJar()
 }
 
 publishing {
@@ -146,12 +135,6 @@ publishing {
                         }
                     }
                 }
-            }
-            artifact(sourceJar) {
-                classifier 'sources'
-            }
-            artifact(javadocJar) {
-                classifier 'javadoc'
             }
 
             if (doSigning) {
@@ -246,9 +229,5 @@ if (doSigning) {
 model {
     tasks.generatePomFileForMavenJavaPublication {
         destination = layout.buildDirectory.file("generated-pom.xml")
-    }
-    tasks.generateMetadataFileForMavenJavaPublication {
-        dependsOn javadocJar
-        dependsOn sourceJar
     }
 }

--- a/build.gradle
+++ b/build.gradle
@@ -1,11 +1,22 @@
 plugins {
     id 'net.researchgate.release' version '2.6.0'
+    id 'org.hidetake.ssh' version '2.11.2'
     id 'java'
     id 'maven-publish'
     id 'signing'
 }
 java {
     sourceCompatibility = JavaVersion.VERSION_17
+}
+
+remotes {
+    rasPi {
+        host = project.findProperty("RasPiHost")
+        user = project.findProperty("RasPiUser")
+        password = project.findProperty("RasPiPassword")
+        fileTransfer = "scp"
+        jschLog = true
+    }
 }
 
 apply plugin: 'java'
@@ -38,6 +49,68 @@ dependencies {
     testImplementation 'junit:junit:4.13.1'
 }
 
+def remoteBuildDir = project.findProperty("RasPiRemoteBuildDir")
+tasks.register('buildNativeOnPi') {
+    description = 'build native lib on remote pi'
+    doLast {
+        ssh.run {
+            session(remotes.rasPi) {
+                println("[buildNativeOnPi] - Copy all files...")
+                put from: './src/', into: remoteBuildDir
+
+                //remove windows line breaks
+                execute "cd $remoteBuildDir; tr -d '\\r' <src/scripts/createNativeLib.sh >src/scripts/createNativeLib.sh"
+                execute "cd $remoteBuildDir; chmod +x src/scripts/createNativeLib.sh"
+                //now build
+                println("[buildNativeOnPi] - Execute build script...")
+                execute "cd $remoteBuildDir; touch logs.txt; bash src/scripts/createNativeLib.sh > logs.txt 2>&1"
+
+                println("[buildNativeOnPi] - Copy generated artifacts back")
+                get from: "$remoteBuildDir/build", into: '.'
+
+                println("[buildNativeOnPi] - All Done! Try ./gradlew assemble")
+            }
+        }
+
+    }
+}
+tasks.register('finalizeBuildNative') {
+    doLast {
+        ssh.run {
+            session(remotes.rasPi) {
+                get from: "$remoteBuildDir/logs.txt", into: 'build/'
+            }
+            println("The captured log output from 'createNativeLib.sh'...")
+            def logFile = project.file("build/logs.txt")
+
+            if (logFile.exists()) {
+                // Read and print the log file contents
+                logFile.eachLine { line ->
+                    println line
+                }
+            } else {
+                println "Log file not found."
+            }
+        }
+
+    }
+}
+
+tasks.buildNativeOnPi.finalizedBy('finalizeBuildNative')
+
+
+tasks.register('checkSharedLibrary') {
+    description = "Check if the shared library exists in the desired location"
+
+    doLast {
+        def libraryFile = file('/build/nativeLib/libws281x.so')
+
+        if (!libraryFile.exists()) {
+            throw new GradleException("Shared library is missing in the desired location.")
+        }
+    }
+}
+
 tasks.register('sourceJar', Jar) {
     from sourceSets.main.allJava
 }
@@ -46,6 +119,8 @@ tasks.register('javadocJar', Jar) {
     dependsOn javadoc
     from javadoc.destinationDir
 }
+
+build.dependsOn checkSharedLibrary
 
 artifacts {
     archives jar
@@ -104,6 +179,7 @@ publishing {
         }
     }
 }
+
 def customizePom(pom) {
     pom.withXml {
         def root = asNode()
@@ -145,6 +221,7 @@ def customizePom(pom) {
         }
     }
 }
+
 model {
     tasks.generatePomFileForMavenJavaPublication {
         destination = layout.buildDirectory.file("generated-pom.xml")

--- a/build.gradle
+++ b/build.gradle
@@ -25,6 +25,10 @@ apply plugin: 'maven-publish'
 apply plugin: 'signing'
 
 group = 'com.github.mbelling'
+//flag to disable signing if keyInfo ain't present
+def doSigning = project.hasProperty('signing.keyId')
+        && project.hasProperty('signing.password')
+        && project.hasProperty('signing.secretKeyRingFile');
 
 sourceSets {
     main {
@@ -97,7 +101,7 @@ tasks.register('checkSharedLibrary') {
     description = "Check if the shared library exists in the desired location"
 
     doLast {
-        def libraryFile = file('/build/nativeLib/libws281x.so')
+        def libraryFile = file('build/nativeLib/libws281x.so')
 
         if (!libraryFile.exists()) {
             throw new GradleException("Shared library is missing in the desired location.")
@@ -120,10 +124,6 @@ artifacts {
     archives jar
     archives sourceJar
     archives javadocJar
-}
-
-signing {
-    sign configurations.archives
 }
 
 publishing {
@@ -149,16 +149,19 @@ publishing {
             artifact(javadocJar) {
                 classifier 'javadoc'
             }
-            // create the signed artifacts
-            project.tasks.signArchives.signatureFiles.each {
-                artifact(it) {
-                    def matcher = it.file =~ /-(sources|javadoc)\.jar\.asc$/
-                    if (matcher.find()) {
-                        classifier = matcher.group(1)
-                    } else {
-                        classifier = null
+
+            if (doSigning) {
+                // create the signed artifacts
+                project.tasks.signArchives.signatureFiles.each {
+                    artifact(it) {
+                        def matcher = it.file =~ /-(sources|javadoc)\.jar\.asc$/
+                        if (matcher.find()) {
+                            classifier = matcher.group(1)
+                        } else {
+                            classifier = null
+                        }
+                        extension = 'jar.asc'
                     }
-                    extension = 'jar.asc'
                 }
             }
         }
@@ -216,14 +219,22 @@ def customizePom(pom) {
     }
 }
 
+// Check if signing properties are defined in gradle.properties
+if (doSigning) {
+    signing {
+        sign configurations.archives
+    }
+
+    tasks.named 'publishMavenJavaPublicationToMavenLocal' {
+        dependsOn project.tasks.named 'signArchives'
+    }
+    tasks.named 'publishMavenJavaPublicationToMavenRepository' {
+        dependsOn project.tasks.named 'signArchives'
+    }
+}
+
 model {
     tasks.generatePomFileForMavenJavaPublication {
         destination = layout.buildDirectory.file("generated-pom.xml")
-    }
-    tasks.publishMavenJavaPublicationToMavenLocal {
-        dependsOn project.tasks.signArchives
-    }
-    tasks.publishMavenJavaPublicationToMavenRepository {
-        dependsOn project.tasks.signArchives
     }
 }

--- a/build.gradle
+++ b/build.gradle
@@ -1,5 +1,11 @@
 plugins {
-  id 'net.researchgate.release' version '2.6.0'
+    id 'net.researchgate.release' version '2.6.0'
+    id 'java'
+    id 'maven-publish'
+    id 'signing'
+}
+java {
+    sourceCompatibility = JavaVersion.VERSION_17
 }
 
 apply plugin: 'java'
@@ -26,19 +32,18 @@ repositories {
 }
 
 dependencies {
-    compile 'org.apache.logging.log4j:log4j-api:2.10.0'
-    runtime 'org.apache.logging.log4j:log4j-core:2.10.0'
+    compileOnly 'org.apache.logging.log4j:log4j-api:2.10.0'
+    runtimeOnly 'org.apache.logging.log4j:log4j-core:2.12.4'
 
-    testCompile 'junit:junit:4.12'
+    testImplementation 'junit:junit:4.13.1'
 }
 
-task sourceJar(type: Jar) {
-    classifier "sources"
+tasks.register('sourceJar', Jar) {
     from sourceSets.main.allJava
 }
 
-task javadocJar(type: Jar, dependsOn: javadoc) {
-    classifier "javadoc"
+tasks.register('javadocJar', Jar) {
+    dependsOn javadoc
     from javadoc.destinationDir
 }
 
@@ -70,10 +75,10 @@ publishing {
                 }
             }
             artifact(sourceJar) {
-                classifier = 'sources'
+                classifier 'sources'
             }
             artifact(javadocJar) {
-                classifier = 'javadoc'
+                classifier 'javadoc'
             }
             // create the signed artifacts
             project.tasks.signArchives.signatureFiles.each {
@@ -142,7 +147,7 @@ def customizePom(pom) {
 }
 model {
     tasks.generatePomFileForMavenJavaPublication {
-        destination = file("$buildDir/generated-pom.xml")
+        destination = layout.buildDirectory.file("generated-pom.xml")
     }
     tasks.publishMavenJavaPublicationToMavenLocal {
         dependsOn project.tasks.signArchives

--- a/build.gradle
+++ b/build.gradle
@@ -122,7 +122,7 @@ tasks.register('checkSharedLibrary') {
         def libraryFile = file('build/nativeLib/libws281x.so')
 
         if (!libraryFile.exists()) {
-            throw new GradleException("Shared library is missing in the desired location.")
+            throw new GradleException("Shared library is missing in the desired location. Consult README and try ./gradlew buildNativeOnPi")
         }
     }
 }

--- a/build.gradle
+++ b/build.gradle
@@ -16,6 +16,7 @@ remotes {
         password = project.findProperty("RasPiPassword")
         fileTransfer = "scp"
         jschLog = true
+        logging = 'slf4j'
     }
 }
 
@@ -52,23 +53,27 @@ dependencies {
 def remoteBuildDir = project.findProperty("RasPiRemoteBuildDir")
 tasks.register('buildNativeOnPi') {
     description = 'build native lib on remote pi'
+    def preamble = "[buildNativeOnPi] -"
+    def scriptName = "src/scripts/createNativeLib.sh"
     doLast {
         ssh.run {
             session(remotes.rasPi) {
-                println("[buildNativeOnPi] - Copy all files...")
-                put from: './src/', into: remoteBuildDir
+                println("$preamble Delete build dir in case it already exists")
+                execute "rm -rf $remoteBuildDir"
+                execute "mkdir $remoteBuildDir"
+                println("$preamble Copy all files...")
+                put from: 'src/', into: "$remoteBuildDir/"
 
                 //remove windows line breaks
-                execute "cd $remoteBuildDir; tr -d '\\r' <src/scripts/createNativeLib.sh >src/scripts/createNativeLib.sh"
-                execute "cd $remoteBuildDir; chmod +x src/scripts/createNativeLib.sh"
+                //execute "cd $remoteBuildDir; tr -d '\\r' < scripts/createNativeLib.sh > scripts/createNativeLib.sh"
+                execute "cd $remoteBuildDir; chmod +x $scriptName"
                 //now build
-                println("[buildNativeOnPi] - Execute build script...")
-                execute "cd $remoteBuildDir; touch logs.txt; bash src/scripts/createNativeLib.sh > logs.txt 2>&1"
-
-                println("[buildNativeOnPi] - Copy generated artifacts back")
+                println("$preamble Execute build script...")
+                execute("cd $remoteBuildDir; bash $scriptName 1>&2")
+                println("$preamble Copy generated artifacts back")
                 get from: "$remoteBuildDir/build", into: '.'
 
-                println("[buildNativeOnPi] - All Done! Try ./gradlew assemble")
+                println("$preamble All Done! Try ./gradlew assemble")
             }
         }
 
@@ -78,18 +83,7 @@ tasks.register('finalizeBuildNative') {
     doLast {
         ssh.run {
             session(remotes.rasPi) {
-                get from: "$remoteBuildDir/logs.txt", into: 'build/'
-            }
-            println("The captured log output from 'createNativeLib.sh'...")
-            def logFile = project.file("build/logs.txt")
 
-            if (logFile.exists()) {
-                // Read and print the log file contents
-                logFile.eachLine { line ->
-                    println line
-                }
-            } else {
-                println "Log file not found."
             }
         }
 

--- a/build.gradle
+++ b/build.gradle
@@ -26,7 +26,7 @@ apply plugin: 'signing'
 
 group = 'com.github.mbelling'
 //flag to disable signing if keyInfo ain't present
-def doSigning = project.hasProperty('signing.keyId')
+boolean doSigning = project.hasProperty('signing.keyId')
         && project.hasProperty('signing.password')
         && project.hasProperty('signing.secretKeyRingFile');
 
@@ -34,11 +34,11 @@ sourceSets {
     main {
         java {
             // Get the generated source from SWIG
-            srcDirs 'build/generatedSource/java'
+            srcDirs file('build/generatedSource/java')
         }
         resources {
             // Get the native .so file
-            srcDirs 'build/nativeLib'
+            srcDirs file('build/nativeLib/')
         }
     }
 }
@@ -134,13 +134,17 @@ publishing {
             artifactId project.name
             version project.version
             from components.java
-            pom.withXml {
-                def pomFile = file("${project.buildDir}/generated-pom.xml")
-                writeTo(pomFile)
-                def pomAscFile = signing.sign(pomFile).signatureFiles[0]
-                artifact(pomAscFile) {
-                    classifier = null
-                    extension = 'pom.asc'
+            pom {
+                withXml {
+                    def pomFile = file("${project.buildDir}/generated-pom.xml")
+                    writeTo(pomFile)
+                    if (doSigning) {
+                        def pomAscFile = signing.sign(pomFile).signatureFiles[0]
+                        artifact(pomAscFile) {
+                            classifier = null
+                            extension = 'pom.asc'
+                        }
+                    }
                 }
             }
             artifact(sourceJar) {
@@ -214,6 +218,12 @@ def customizePom(pom) {
                 developer {
                     name "Matthew Bellinger"
                 }
+                developer {
+                    name "MNG"
+                    roles {
+                        role "Update and Improvements like added checks, examples and increased usability"
+                    }
+                }
             }
         }
     }
@@ -236,5 +246,9 @@ if (doSigning) {
 model {
     tasks.generatePomFileForMavenJavaPublication {
         destination = layout.buildDirectory.file("generated-pom.xml")
+    }
+    tasks.generateMetadataFileForMavenJavaPublication {
+        dependsOn javadocJar
+        dependsOn sourceJar
     }
 }

--- a/build.gradle
+++ b/build.gradle
@@ -13,7 +13,11 @@ remotes {
     rasPi {
         host = project.findProperty("RasPiHost")
         user = project.findProperty("RasPiUser")
-        password = project.findProperty("RasPiPassword")
+        if(project.hasProperty("RasPiKey")){
+            identity = file(project.findProperty("RasPiKey"))
+        }else {
+            password = project.findProperty("RasPiPassword")
+        }
         fileTransfer = "scp"
         jschLog = true
         logging = 'slf4j'
@@ -53,6 +57,21 @@ dependencies {
     testImplementation 'junit:junit:4.13.1'
 }
 
+tasks.register('installSwigOnPi'){
+    description = 'try to install swig on the raspi'
+    doLast{
+        ssh.run{
+            session(remotes.rasPi){
+                println("ATEMPTING TO INSTALL SWIG")
+                put (from:'install-swig-on-pi.sh', into: '~')
+                execute "cd ~; chmod +x install-swig-on-pi.sh"
+                execute("cd ~; bash install-swig-on-pi.sh 1>&2") //re-route stdout to stderr so the user sees all the log output
+                println("SUCCESS!")
+            }
+        }
+    }
+}
+
 def remoteBuildDir = project.findProperty("RasPiRemoteBuildDir")
 tasks.register('buildNativeOnPi') {
     description = 'build native lib on remote pi'
@@ -72,7 +91,7 @@ tasks.register('buildNativeOnPi') {
                 execute "cd $remoteBuildDir; chmod +x $scriptName"
                 //now build
                 println("$preamble Execute build script...")
-                execute("cd $remoteBuildDir; bash $scriptName 1>&2")
+                execute("cd $remoteBuildDir; bash $scriptName 1>&2") //re-route stdout to stderr so the user sees all the log output
                 println("$preamble Copy generated artifacts back")
                 get from: "$remoteBuildDir/build", into: '.'
 

--- a/build.gradle
+++ b/build.gradle
@@ -59,13 +59,15 @@ dependencies {
 
 tasks.register('installSwigOnPi'){
     description = 'try to install swig on the raspi'
+    def swigDir = "~/rpi_ws281x_build/swigInstall"
     doLast{
         ssh.run{
             session(remotes.rasPi){
                 println("ATEMPTING TO INSTALL SWIG")
-                put (from:'install-swig-on-pi.sh', into: '~')
-                execute "cd ~; chmod +x install-swig-on-pi.sh"
-                execute("cd ~; bash install-swig-on-pi.sh 1>&2") //re-route stdout to stderr so the user sees all the log output
+                execute "mkdir -p $swigDir"
+                put (from:'install-swig-on-pi.sh', into: swigDir)
+                execute "cd $swigDir; chmod +x install-swig-on-pi.sh"
+                execute("cd $swigDir; bash install-swig-on-pi.sh 1>&2") //re-route stdout to stderr so the user sees all the log output
                 println("SUCCESS!")
             }
         }
@@ -80,9 +82,8 @@ tasks.register('buildNativeOnPi') {
     doLast {
         ssh.run {
             session(remotes.rasPi) {
-                println("$preamble Delete build dir in case it already exists")
-                execute "rm -rf $remoteBuildDir"
-                execute "mkdir $remoteBuildDir"
+                println("$preamble make build dir in case it doesn't exists")
+                execute "mkdir -p $remoteBuildDir"
                 println("$preamble Copy all files...")
                 put from: 'src/', into: "$remoteBuildDir/"
 

--- a/build.gradle
+++ b/build.gradle
@@ -1,38 +1,27 @@
 plugins {
     id 'net.researchgate.release' version '2.6.0'
+    id 'org.hidetake.ssh' version '2.11.2'
     id 'java'
     id 'maven-publish'
     id 'signing'
+}
+java {
+    sourceCompatibility = JavaVersion.VERSION_17
+}
+
+remotes {
+    rasPi {
+        host = project.findProperty("RasPiHost")
+        user = project.findProperty("RasPiUser")
+        password = project.findProperty("RasPiPassword")
+        fileTransfer = "scp"
+        jschLog = true
+    }
 }
 
 apply plugin: 'java'
 apply plugin: 'maven-publish'
 apply plugin: 'signing'
-
-java {
-    sourceCompatibility = JavaVersion.VERSION_17
-}
-repositories {
-    mavenCentral()
-}
-
-configurations { sshAntTask }
-dependencies {
-    compileOnly 'org.apache.logging.log4j:log4j-api:2.10.0'
-    runtimeOnly 'org.apache.logging.log4j:log4j-core:2.12.4'
-    sshAntTask 'org.apache.ant:ant-jsch:1.10.14'
-    testImplementation 'junit:junit:4.13.1'
-}
-
-ant.taskdef(
-        name: 'scp',
-        classname: 'org.apache.tools.ant.taskdefs.optional.ssh.Scp',
-        classpath: configurations.sshAntTask.asPath)
-
-ant.taskdef(
-        name: 'ssh',
-        classname: 'org.apache.tools.ant.taskdefs.optional.ssh.SSHExec',
-        classpath: configurations.sshAntTask.asPath)
 
 group = 'com.github.mbelling'
 
@@ -48,71 +37,66 @@ sourceSets {
         }
     }
 }
-def remBuildDir = project.findProperty("RasPiRemoteBuildDir")
 
+repositories {
+    mavenCentral()
+}
+
+dependencies {
+    compileOnly 'org.apache.logging.log4j:log4j-api:2.10.0'
+    runtimeOnly 'org.apache.logging.log4j:log4j-core:2.12.4'
+
+    testImplementation 'junit:junit:4.13.1'
+}
+
+def remoteBuildDir = project.findProperty("RasPiRemoteBuildDir")
 tasks.register('buildNativeOnPi') {
     description = 'build native lib on remote pi'
-    def host = project.findProperty("RasPiHost")
-    def un = project.findProperty("RasPiUser")
-    def pw = project.findProperty("RasPiPassword")
     doLast {
-        println("trying to log in to $un@$host with pw $pw")
-        ant.scp(
-                file: "src/ $un@$host:$remBuildDir",
-                password: $pw
-                host: host,
-                username: un,
-                password: pw,
-                verbose: true,
-                failonerror: true
-        )
-        ant.ssh(
-                host: host,
-                username: un,
-                password: pw,
-                command: "sudo bash $remBuildDir/src/scripts/createNativeLib.sh",
-                trust: 'yes',
-                verbose: true,
-                outputproperty: "outProp",
-                resultproperty: "resProp",
-                failonerror: true
-        )
+        ssh.run {
+            session(remotes.rasPi) {
+                println("[buildNativeOnPi] - Copy all files...")
+                put from: './src/', into: remoteBuildDir
+
+                //remove windows line breaks
+                execute "cd $remoteBuildDir; tr -d '\\r' <src/scripts/createNativeLib.sh >src/scripts/createNativeLib.sh"
+                execute "cd $remoteBuildDir; chmod +x src/scripts/createNativeLib.sh"
+                //now build
+                println("[buildNativeOnPi] - Execute build script...")
+                execute "cd $remoteBuildDir; touch logs.txt; bash src/scripts/createNativeLib.sh > logs.txt 2>&1"
+
+                println("[buildNativeOnPi] - Copy generated artifacts back")
+                get from: "$remoteBuildDir/build", into: '.'
+
+                println("[buildNativeOnPi] - All Done! Try ./gradlew assemble")
+            }
+        }
+
+    }
+}
+tasks.register('finalizeBuildNative') {
+    doLast {
+        ssh.run {
+            session(remotes.rasPi) {
+                get from: "$remoteBuildDir/logs.txt", into: 'build/'
+            }
+            println("The captured log output from 'createNativeLib.sh'...")
+            def logFile = project.file("build/logs.txt")
+
+            if (logFile.exists()) {
+                // Read and print the log file contents
+                logFile.eachLine { line ->
+                    println line
+                }
+            } else {
+                println "Log file not found."
+            }
+        }
+
     }
 }
 
-tasks.register("finalizeBuild") {
-    doLast {
-
-        println("**********************************************************************")
-        println("                         LOG from remote build:")
-        println("**********************************************************************")
-        println(ant.getProperty("outProp"))
-    }
-}
-buildNativeOnPi.finalizedBy finalizeBuild
-
-
-/*
-        println("[buildNativeOnPi] - Copy all files...")
-        commandLine 'scp'
-        put from: './src/', into: remoteBuildDir
-
-        //remove windows line breaks
-        execute "cd $remoteBuildDir; tr -d '\\r' <src/scripts/createNativeLib.sh >src/scripts/createNativeLib.sh"
-        execute "cd $remoteBuildDir; chmod +x src/scripts/createNativeLib.sh"
-        //now build
-        println("[buildNativeOnPi] - Execute build script...")
-        execute "cd $remoteBuildDir; touch logs.txt; bash src/scripts/createNativeLib.sh > logs.txt 2>&1"
-
-        println("[buildNativeOnPi] - Copy generated artifacts back")
-        get from: "$remoteBuildDir/build", into: '.'
-
-        println("[buildNativeOnPi] - All Done! Try ./gradlew assemble")
-
-    }
-}*/
-
-tasks.register('copyFilesBackAfterRemoteBuild')
+tasks.buildNativeOnPi.finalizedBy('finalizeBuildNative')
 
 
 tasks.register('checkSharedLibrary') {

--- a/gradle.properties
+++ b/gradle.properties
@@ -7,7 +7,9 @@ sonatypePassword=empty
 #example
 #ssh pi@raspi
 #pi's pw: hithere
+#if you want to use a key instead of a password, uncomment the RasPiKey property
 RasPiUser=pi
 RasPiHost=raspi
 RasPiPassword=hithere
+#RasPiKey=C:\\Path\\To\\Private\\SSH\\key for windows or /path/to/private/ssh/key for unix
 RasPiRemoteBuildDir=~/rpi_ws281x_build

--- a/gradle.properties
+++ b/gradle.properties
@@ -3,3 +3,11 @@ version=2.0.0-SNAPSHOT
 # Dummy values so the build does not break
 sonatypeUsername=empty
 sonatypePassword=empty
+#Raspi remote build properties
+#example
+#ssh pi@raspi
+#pi's pw: hithere
+RasPiUser=pi
+RasPiHost=raspi
+RasPiPassword=hithere
+RasPiRemoteBuildDir=~/rpi_ws281x_build

--- a/gradle/wrapper/gradle-wrapper.properties
+++ b/gradle/wrapper/gradle-wrapper.properties
@@ -1,6 +1,5 @@
-#Fri Feb 23 09:41:07 EST 2018
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-4.5.1-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-8.4-bin.zip

--- a/install-swig-on-pi.sh
+++ b/install-swig-on-pi.sh
@@ -2,11 +2,13 @@
 set -e
 set -o pipefail
 cd ~
-#sudo apt-get --yes --force-yes install build-essential libpcre2-dev libpcre3-dev
+sudo apt-get --yes --force-yes install build-essential libpcre2-dev libpcre3-dev
+set +e
 echo "check if java is already installed..."
 java --version | grep jdk
 if [ $? -ne 0 ]
 then
+  set -e
    echo "didn't detect install, will try..."
    echo "check cpu architecture..."
    cat /proc/cpuinfo | grep ARMv6
@@ -27,7 +29,7 @@ then
 else
   echo "detected java, skipping install"
 fi
-
+set -e
 mkdir -p ~/rpi_ws281x_build/swigInstall
 cd ~/rpi_ws281x_build/swigInstall
 if [ -d "swig-4.1.1" ]; then
@@ -73,11 +75,14 @@ if [ "$FOUNDJNI" = "true" ]; then
   sudo ./configure
 else
   if [ -z ${SELFJNI+x} ]; then
+    echo "*********************[install-swig-on-pi.sh]*********************"
     echo "NO JNI FOUND! NO POINT IN COMPILING SWIG ONLY FOR IT TO FAIL!"
     echo "you could try running ./configure in $(pwd) yourself. If at the end it outputs:"
     echo "'The SWIG test-suite and examples are configured for the following languages:'"
     echo "    java perl5"
     echo "...then you're fine. You can then run the very time consuming command 'make' to compile swig."
+    echo "*****************************************************************"
+
     exit 1
   else
     echo "passing include dir '$SELFJNI' to ./configure..."
@@ -85,16 +90,28 @@ else
   fi
 fi
 sudo make
+echo "*********************[install-swig-on-pi.sh]*********************"
 echo "COMPILATION DONE!!"
+echo "*****************************************************************"
 cd Examples/java/class && make check
+echo "*********************[install-swig-on-pi.sh]*********************"
 echo "CHECK DONE!!"
+echo "*****************************************************************"
 cd ../../../
 sudo make install
-swig -version | grep "SWIG Version 4.1.1"
+set +e
+source ~/.bashrc
+/usr/local/bin/swig -version | grep "SWIG Version 4.1.1"
 if [ $? -ne 0 ]; then
+  echo "*********************[install-swig-on-pi.sh]*********************"
   echo "swig -version command didn't output expected info. Install probably failed 😣"
+  echo "*****************************************************************"
+
   exit 1;
 else
+  echo "*********************[install-swig-on-pi.sh]*********************"
   echo "Install worked!!! 🔥🔥😀"
+  echo "*****************************************************************"
+
   exit 0;
 fi

--- a/install-swig-on-pi.sh
+++ b/install-swig-on-pi.sh
@@ -2,18 +2,96 @@
 set -e
 set -o pipefail
 cd ~
-sudo apt-get --yes --force-yes install build-essential libpcre2-dev libpcre3-dev
-sudo apt-get --yes --force-yes  install default-jdk-headless
-wget http://prdownloads.sourceforge.net/swig/swig-4.1.1.tar.gz
-tar -xzvf swig-4.1.1.tar.gz
+#sudo apt-get --yes --force-yes install build-essential libpcre2-dev libpcre3-dev
+echo "check if java is already installed..."
+java --version | grep jdk
+if [ $? -ne 0 ]
+then
+   echo "didn't detect install, will try..."
+   echo "check cpu architecture..."
+   cat /proc/cpuinfo | grep ARMv6
+   if [ $? -eq 0 ]
+   then
+     echo "DETECTED ARMv6, WILL TRY AND INSTALL ZULU JDK!"
+     mkdir -p /usr/lib/jvm
+     cd /usr/lib/jvm
+     sudo wget https://cdn.azul.com/zulu-embedded/bin/zulu11.41.75-ca-jdk11.0.8-linux_aarch32hf.tar.gz
+     sudo tar -xzvf zulu11.41.75-ca-jdk11.0.8-linux_aarch32hf.tar.gz
+     sudo rm zulu11.41.75-ca-jdk11.0.8-linux_aarch32hf.tar.gz
+     sudo update-alternatives --install /usr/bin/java java /usr/lib/jvm/zulu11.41.75-ca-jdk11.0.8-linux_aarch32hf/bin/java 1
+     sudo update-alternatives --install /usr/bin/javac javac /usr/lib/jvm/zulu11.41.75-ca-jdk11.0.8-linux_aarch32hf/bin/javac 1
+   else
+     echo "didn't detect ARMv6, will try and install default jdk..."
+     sudo apt-get --yes --force-yes  install default-jdk-headless
+   fi
+else
+  echo "detected java, skipping install"
+fi
+
+mkdir -p ~/rpi_ws281x_build/swigInstall
+cd ~/rpi_ws281x_build/swigInstall
+if [ -d "swig-4.1.1" ]; then
+    echo "found swig folder, skipping download..."
+else
+    echo "attempting swig 4.1.1 download"
+    wget http://prdownloads.sourceforge.net/swig/swig-4.1.1.tar.gz
+    tar -xzvf swig-4.1.1.tar.gz
+    rm swig-4.1.1.tar.gz
+fi
+
 cd swig-4.1.1
-./configure && make
+echo "trying to figure out if --with-javainc arg needs/can to be passed to swig configure..."
+echo "executing part of swig configure script to determine if it will find jni.h..."
+
+JAVAINCDIR="/usr/j2sdk*/include /usr/local/j2sdk*/include /usr/jdk*/include /usr/local/jdk*/include /opt/j2sdk*/include /opt/jdk*/include /usr/java/include /usr/java/j2sdk*/include /usr/java/jdk*/include /usr/local/java/include /opt/java/include /usr/include/java /usr/local/include/java /usr/lib/java/include /usr/lib/jvm/java*/include /usr/lib64/jvm/java*/include /usr/include/kaffe /usr/local/include/kaffe /usr/include"
+
+
+for d in $JAVAINCDIR ; do
+  if test -r "$d/jni.h" ; then
+    echo "yep, found something, setting flag"
+	  FOUNDJNI="true"
+	  break
+  fi
+done
+
+if [ "$FOUNDJNI" != "true" ]; then
+  echo "configure won't find anything, will try to get a path for JNI.h"
+  for d2 in /usr/lib/jvm/*/include ; do
+    if test -r "$d2/jni.h" ; then
+      echo "SUCCESS! found $d2"
+      echo "will pass to configure"
+  	  SELFJNI=$d2
+  	  break
+    fi
+  done
+fi
+
+
+
+
+if [ "$FOUNDJNI" = "true" ]; then
+  sudo ./configure
+else
+  if [ -z ${SELFJNI+x} ]; then
+    echo "NO JNI FOUND! NO POINT IN COMPILING SWIG ONLY FOR IT TO FAIL!"
+    echo "you could try running ./configure in $(pwd) yourself. If at the end it outputs:"
+    echo "'The SWIG test-suite and examples are configured for the following languages:'"
+    echo "    java perl5"
+    echo "...then you're fine. You can then run the very time consuming command 'make' to compile swig."
+    exit 1
+  else
+    echo "passing include dir '$SELFJNI' to ./configure..."
+    sudo ./configure --with-javaincl=$SELFJNI
+  fi
+fi
+sudo make
+echo "COMPILATION DONE!!"
 cd Examples/java/class && make check
+echo "CHECK DONE!!"
 cd ../../../
 sudo make install
 swig -version | grep "SWIG Version 4.1.1"
-if [ $? -ne 0 ]
-then
+if [ $? -ne 0 ]; then
   echo "swig -version command didn't output expected info. Install probably failed 😣"
   exit 1;
 else

--- a/install-swig-on-pi.sh
+++ b/install-swig-on-pi.sh
@@ -1,0 +1,22 @@
+#!/bin/bash
+set -e
+set -o pipefail
+cd ~
+sudo apt-get --yes --force-yes install build-essential libpcre2-dev libpcre3-dev
+sudo apt-get --yes --force-yes  install default-jdk-headless
+wget http://prdownloads.sourceforge.net/swig/swig-4.1.1.tar.gz
+tar -xzvf swig-4.1.1.tar.gz
+cd swig-4.1.1
+./configure && make
+cd Examples/java/class && make check
+cd ../../../
+sudo make install
+swig -version | grep "SWIG Version 4.1.1"
+if [ $? -ne 0 ]
+then
+  echo "swig -version command didn't output expected info. Install probably failed 😣"
+  exit 1;
+else
+  echo "Install worked!!! 🔥🔥😀"
+  exit 0;
+fi

--- a/src/main/java/com/github/mbelling/ws281x/Ws281xLedStrip.java
+++ b/src/main/java/com/github/mbelling/ws281x/Ws281xLedStrip.java
@@ -1,8 +1,8 @@
 package com.github.mbelling.ws281x;
 
 import com.github.mbelling.ws281x.jni.*;
-import org.apache.logging.log4j.LogManager;
-import org.apache.logging.log4j.Logger;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 import java.io.IOException;
 import java.nio.file.Files;
@@ -16,7 +16,7 @@ import static com.github.mbelling.ws281x.Color.buildColour;
  * Basic class to interface with the rpi_ws281x native library
  */
 public class Ws281xLedStrip implements LedStrip {
-    private static final Logger LOG = LogManager.getLogger( Ws281xLedStrip.class );
+    private static final Logger LOG = LoggerFactory.getLogger( Ws281xLedStrip.class );
     private static final String LIB_NAME = "ws281x";
 
     private static final AtomicBoolean loaded = new AtomicBoolean( false );

--- a/src/scripts/createNativeLib.sh
+++ b/src/scripts/createNativeLib.sh
@@ -10,6 +10,8 @@
 # Configuration
 # *********************************************************************************************************************
 
+#This is needed since somehow the swig command isn't being found on my RasPi 0 otherwise
+PATH=$PATH:/usr/local/bin/
 # This defines the JDK to use for JNI header files; automatically picks first dir using ls
 JDK_PATH="/usr/lib/jvm"
 JDK_DIR=$(ls "${JDK_PATH}" | head -n 1)
@@ -63,9 +65,11 @@ function programInstalled
 # *********************************************************************************************************************
 # Main
 # *********************************************************************************************************************
-
-echo "NeoPixel ws281x Library Compiler"
-echo "****************************************************"
+echo "**********************[createNativeLib.sh]******************************"
+echo "*                                                                      *"
+echo "*               NeoPixel ws281x Library Compiler                       *"
+echo "*                                                                      *"
+echo "************************************************************************"
 
 # Check dependencies installed
 set -e

--- a/src/scripts/createNativeLib.sh
+++ b/src/scripts/createNativeLib.sh
@@ -69,11 +69,12 @@ echo "****************************************************"
 
 # Check dependencies installed
 set -e
-programInstalled "swig -version" "SWIG Version" "Error - SWIG is not installed, cannot continue!" "Check - SWIG installed..."
-programInstalled "gcc --version" "free software" "Error - GCC is not installed, cannot continue!" "Check - GCC installed..."
-programInstalled "ar --version" "free software" "Error - AR is not installed, cannot continue!" "Check - AR installed..."
-programInstalled "ranlib -v" "free software" "Error - ranlib is not installed, cannot continue!" "Check - ranlib installed..."
-programInstalled "git --version" "git version" "Error - git is not installed, cannot continue!" "Check - git installed..."
+programInstalled "swig -version" "SWIG Version" "Error - SWIG is not installed, cannot continue! Check out the tutorial in this repo's README!" "✅ - SWIG installed..."
+programInstalled "java --version" "Runtime Environment" "Error  -  Java is not installed, cannot continue! (SWIG won't work either w/out java btw... Check out the tutorial in the README!)"
+programInstalled "gcc --version" "free software" "Error - GCC is not installed, cannot continue!" "✅ - GCC installed..."
+programInstalled "ar --version" "free software" "Error - AR is not installed, cannot continue!" "✅ - AR installed..."
+programInstalled "ranlib -v" "free software" "Error - ranlib is not installed, cannot continue!" "✅ - ranlib installed..."
+programInstalled "git --version" "git version" "Error - git is not installed, cannot continue!" "✅ - git installed..."
 set +e
 
 # Clean workspace

--- a/src/scripts/createNativeLib.sh
+++ b/src/scripts/createNativeLib.sh
@@ -89,10 +89,10 @@ rm -rf build
 echo "Cloning rpi_ws281x repository..."
 git clone https://github.com/jgarff/rpi_ws281x.git ${NATIVE_SRC}
 
-# At the time of this writing this repository does not tag versions, so checking out at a specific commit so we build a consistent library
+# checking out v1.0.0 so we build a consistent library
 echo "Checking out specific revision..."
 pushd ${NATIVE_SRC}
-git checkout 6851d9fb090f8a4703d2ceac97da2de617b09e8d
+git checkout v1.0.0
 popd
 
 # Create all the required dirs

